rpcserver.go

```diff
@@ -1934,6 +1934,35 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 
 	localFundingAmt := btcutil.Amount(in.LocalFundingAmount)
 	remoteInitialBalance := btcutil.Amount(in.PushSat)
+
+	// If the FundMax flag is set, ensure that the acceptable minimum local
+	// amount adheres to the amount to be pushed to the remote, and to
+	// current rules, while also respecting the settings for the maximum
+	// channel size.
+	var fundMinAmount, fundUpToMaxAmt btcutil.Amount
+	if in.FundMax {
+		// We assume the configured maximum channel size to be the upper
+		// bound of our "maxed" out funding attempt.
+		fundUpToMaxAmt = btcutil.Amount(r.cfg.MaxChanSize)
+
+		// Since the standard non-fundmax flow requires the minimum
+		// funding amount to be at least in the amount of the initial
+		// remote balance(push amount) we need to adjust the minimum
+		// funding amount accordingly. We initially assume the minimum
+		// allowed channel size as minimum funding amount.
+		fundMinAmount = funding.MinChanFundingSize
+
+		// If fundMinAmount is less than the initial remote balance we
+		// simply assign the initial remote balance to fundMinAmount in
+		// order to fullfil the criterion. Whether or not this so
+		// determined minimum amount is actually available is
```

Comment:
question: does available mean here, whether we have enough funds in the wallet when checking it downstream?

Reply:
Yes that's right. We don't check in the server what our actual wallet balance is but just set the boundaries of our channel size. The coin selection flow then determines if we have sufficient funds.